ShowMyCursedRealm/threads.aadl

```diff
@@ -0,0 +1,212 @@
+package threads
+public
+	with Data_Model;
+
+	-----Data
+	data Video_Stream
+		properties
+			Data_Model::Data_Representation => integer;
+	end Video_Stream;
+	
+	data Window_State
+		properties
+			Data_Model::Data_Representation => integer;
+	end Window_State;
+	
+	data Door_State
+		properties
+			Data_Model::Data_Representation => integer;
+	end Door_State;
+	
+	data Pin
+		properties
+			Data_Model::Data_Representation => integer;
+	end Pin;
+	
+	data BooleanValue
+		properties
+			Data_Model::Data_Representation => integer;
+	end BooleanValue;
+	
+	data Timestamp
+		properties
+			Data_Model::Data_Representation => integer;
+	end Timestamp;
+	----------------------
+	
+	-----Subprograms
+	subprogram Video_Processing_Spg
+		features
+			video_stream: in parameter Video_stream;
+		properties
+			source_language => (C);
+			source_name => "room_controller_video_thread";
+			source_text => ("room.c");
+	end Video_Processing_Spg;
+	
+	subprogram Window_Sensor_Spg
+		features
+			window_state: in parameter Window_State;
+		properties
+			source_language => (C);
+			source_name => "room_controller_video_thread";
+			source_text => ("room.c");
+	end Window_Sensor_Spg;
+	
+	subprogram Door_Sensor_Spg
+		features
+			door_state: in parameter Door_State;
+			last_correct_pin_time: in parameter Timestamp;
+		properties
+			source_language => (C);
+			source_name => "room_controller_video_thread";
+			source_text => ("room.c");
+	end Door_Sensor_Spg;
+	
+	subprogram Movement_Sensor_Spg
+		features
+			movement_data: in parameter BooleanValue;
+		properties
+			source_language => (C);
+			source_name => "room_controller_video_thread";
+			source_text => ("room.c");
+	end Movement_Sensor_Spg;
+	
+	subprogram Pin_Device_Spg
+		features
+			entered_pin: in parameter Pin;
+			correct_pin_timestamp: out parameter Timestamp;
+			tries_exceeded: out parameter BooleanValue;
+			
+		properties
+			source_language => (C);
+			source_name => "room_controller_video_thread";
+			source_text => ("room.c");
+	end Pin_Device_Spg;
+
+	
+	subprogram Alarm_Spg
+		properties
+			source_language => (C);
+			source_name => "room_controller_video_thread";
+			source_text => ("room.c");
+	end Alarm_Spg;
+
+	-------------------
+
+
+	-----Threads
+  	thread video_thread
+		features
+			video_stream: in data port Video_Stream;
+	end video_thread;
+
+	thread implementation video_thread.i
+		calls
+			Mycalls: {
+				Q_Spg: subprogram Video_Processing_Spg;
+			};
+		connections
+			c0: parameter video_stream -> Q_Spg.video_stream;
+		properties
+			Dispatch_Protocol => Sporadic;
```

Comment:
Tutaj też myślę, że powinien być `Periodic`, ponieważ z tego co rozumiem, to powinno działać tak, że co jakiś czas my sprawdzamy kamerę i wysyłamy obraz z tej kamery na controller.